scriptmodules/supplementary/emulationstation/configscripts/reicast.sh

```diff
@@ -0,0 +1,159 @@
+        start)
+            keys=("Btn_Start")
+            ;;
+        select)
+            keys=("Quit")
+            ;;
+        leftanalogleft)
+            keys=("Axis_X")
+            ;;
+        leftanalogright)
+            keys=("Axis_X")
+            ;;
+        leftanalogup)
+            keys=("Axis_Y")
+            ;;
+        leftanalogdown)
+            keys=("Axis_Y")
+            ;;
+        *)
+            return
+            ;;
+    esac
+
+    local key
+    local value
+    for key in "${keys[@]}"; do
+        # read key value. Axis takes two key/axis values.
+        case "$input_type" in
+            axis) 
+                # key "X/Y Axis" needs different button naming
+                if [[ "$key" == "DPad2_Left" ]] ; then
+                    iniSet "button.$input_id" "Axis_LT"
+                elif [[ "$key" == "DPad2_Right" ]] ; then
+                    iniSet "button.$input_id" "Axis_RT"
+                elif [[ "$key" == "DPad_Up" || "$key" == "DPad_Down" ]]; then
+                    iniSet "axis.$input_id" "Axis_Y" 
+                elif [[ "$key" == "DPad_Right" || "$key" == "DPad_Left" ]]; then
+                    iniSet "axis.$input_id" "Axis_X" 
+                elif [[ "$key" == *Axis* ]] ; then
+                    iniSet "axis.$input_id" "${key}" 
+                fi
+                ;;
+            hat)
+                ;;
+            *)
+                if [[ "$key" == "Axis_LT" ]] ; then
+                    iniSet "button.$input_id" "DPad2_Left"
+                elif [[ "$key" == "Axis_RT" ]] ; then
+                    iniSet "button.$input_id" "DPad2_Right"
+                elif [[ "$key" == "Axis_X" || "$key" == "Axis_Y" ]] ; then
+                else
+                    iniSet "button.$input_id" "$key"
+                fi
+                ;;
+        esac
+    done
+}
+
+function onend_reicast_joystick() {
+    local device_type=$1
+    local device_name=$2
+
+    echo "; ${device_name}_END " >> /tmp/reicasttempconfig.cfg
+    echo "" >> /tmp/reicasttempconfig.cfg
+
+    # abort if old device config cannot be deleted.
+    local file="/home/pi/.reicast/emu.cfg"
+    if [[ ! -d "/home/pi/.reicast" ]]; then
+        mkdir /home/pi/.reicast
+    fi
+    if [[ -f "$file" ]]; then
+        # backup current config file
+        cp "$file" "${file}.bak"
+        # if reicast did not run frames are there
+        if grep -q "${device_name}_END" "$file" ; then
+            sed -i /"${device_name}_START"/,/"${device_name}_END"/d "$file"
+        # reicast removes frames after first run but adds axis.31
+        else
+            sed -i /"${device_name}"/,/"axis.31="/d "$file"
+        fi
+        if grep -q "$device_name" "$file" ; then
+            rm /tmp/reicasttempconfig.cfg
+            return
+        fi
+    fi
+
+    # read temp device configuration and append to emu.cfg
+    local deviceconfig=`cat /tmp/reicasttempconfig.cfg`
+    echo "$deviceconfig" >> "$file"
+    rm /tmp/reicasttempconfig.cfg
```

Comment:
`cat /tmp/reicasttempconfig.cfg >> "$file"` (I made this change in your other inputconfig also)